### DISCUSSION TITLE
Data Type Archive Manipulation

### DISCUSSION THREAD

**Discussion by bdemick** [NationalSecurityAgency/ghidra#5187] [Q&A]
I'm attempting to post-process a data type archive that ended up with a bunch of `.conflict` function signatures that are correct by removing the "bad" data type, and setting the name of the matching .conflict signature to reflect the correct name. However, I keep getting `DuplicateNameException`. After several iterations, I've landed here:

```
def fix_conflicts(data_type_manager):
    # For each function that ends in .conflict, find the corresponding "untyped" function, delete that one,
    # and rename the conflict function to be the untyped name
    all_types = list(data_type_manager.getAllDataTypes())
    conflicts = [typ for typ in all_types if typ.name.endswith(".conflict")]

    for con in conflicts:
        stripped_con = os.path.splitext(con.name)[0]
        for dtype in all_types:
            if dtype.name == stripped_con:
                # Remove the original function from the gdt and rename the .conflict
                tid = data_type_manager.startTransaction("remove {}".format(dtype.name))
                data_type_manager.remove(dtype, monitor)
                data_type_manager.endTransaction(tid, True)
                tid = data_type_manager.startTransaction("rename {} to {}".format(con.name, dtype.name))
                con.setName(stripped_con)
                data_type_manager.endTransaction(tid, True)
                break
```
I feel like I'm missing something subtle here. I've tried inserting `data_type_manager.save()` between the transactions, but that seems to only slow things down to a crawl with the same end result. Any ideas what I'm missing?

**Comment by astrelsky** [comment]
Try using `data_type_manager.replaceDataType(DataType existingDt, DataType replacementDt, boolean updateCategoryPath)` and then setting the name.

Just watch out for overloaded c++ function signatures because they will both be correct but will cause a conflict because they have the same name.
